src/runtime/HalideRuntime.h

```diff
@@ -475,21 +474,60 @@ typedef struct buffer_t {
     * replaced with a more general type code in the future. */
     int32_t elem_size;
 
+    /**
+    * Flag bits from halide_buffer_flag_bits_t.
+    */
+    int32_t flags;
+
+    /** An unused field, used to ensure that the struct is an even multiple
+     * of 64 bits on 32-bit systems. */
+    void* _unused;
+} buffer_t;
+
+#endif
+
+#if defined(__cplusplus) && (__cplusplus > 199711L || _MSC_VER >= 1800)
+// Let's assert specific size expectations, because disagreement
+// between caller and callee produces suboptimal results.
+static_assert(sizeof(buffer_t) == (sizeof(void*) == 4 ? 72 : 80),
+              "sizeof(buffer_t) is incorrect");
+#endif
+
+enum halide_buffer_flag_bits_t {
     /** This should be true if there is an existing device allocation
     * mirroring this buffer, and the data has been modified on the
     * host side. */
-    bool host_dirty;
+    halide_buffer_host_dirty = (1 << 0),
 
     /** This should be true if there is an existing device allocation
     mirroring this buffer, and the data has been modified on the
     device side. */
-    bool dev_dirty;
-
-    uint8_t _padding[2];
-} buffer_t;
-#pragma pack(pop)
+    halide_buffer_dev_dirty = (1 << 1)
+};
 
-#endif
+inline bool halide_buffer_get_host_dirty(const buffer_t* buf) {
+    return (buf->flags & halide_buffer_host_dirty) != 0;
+}
+
+inline bool halide_buffer_get_dev_dirty(const buffer_t* buf) {
```

Comment:
Given the issue we had this morning with inline on os x, make sure you test apps/glsl